File: PHP/ValidateStackSequences.php
<?php
/**
 * @version PHP 8.2.0
 * @author MARIO SAM <eu@mariosam.com.br>
 * @see I would love to work with you instead solving web code tests: hire me!
 */
namespace PHP;

class ValidateStackSequences {    

    /**
     * @param Integer[] $pushed
     * @param Integer[] $popped
     * @return Boolean
     */
    function validateStackSequences($pushed, $popped) {
        if (count($pushed) <= 1) return true;
        $stack = [];
        $popIndex = 0;

        for ($i=0; $i < count($pushed); $i++) {
            array_push($stack,$pushed[$i]);
            while (count($stack) && $stack[count($stack) - 1] == $popped[$popIndex]) {
                array_pop($stack);
                $popIndex++;
            }
        }

        return count($stack) == 0;
    }
}

File: JAVA/ValidateStackSequences.java
/**
 * @version JAVA
 * @author MARIO SAM <eu@mariosam.com.br>
 * @see I would love to work with you instead solving web code tests: hire me!
 */
package JAVA;

import java.util.Stack;

public class ValidateStackSequences {

    public static void main(String[] args) {
        System.out.printf("Resultado: %d\n", validateStackSequences( new int[]{1,2,3,4,5}, new int[]{5,4,3,2,1} ));
    }

    public static boolean validateStackSequences(int[] pushed, int[] popped) {
        Stack<Integer> st = new Stack<>();
        int it = 0;

        for (int i : pushed) {
            if (i == popped[it]) {
                ++it;
                while (!st.isEmpty() && st.peek() == popped[it]) {
                    st.pop();
                    it++;
                }
            } else {
                st.push(i);
            }
        }
				
        return st.isEmpty();
    }

}
